### PR TITLE
documentation: move 32/64/native literals out of the extension chapter

### PR DESCRIPTION
**Pull request by Octachron**
This PR proposes to move the documentation for 32-bit, 64-bit and native integer literals to the main part of the manual. More precisely,  the description of these literals in `refman/exten.etex` is moved to `refman/const.etex` and `refman/lex.etex`. Moreover, the description of  `int32`, `int64` and `nativeint` now mentions the existence of those literals in both the standard library documentation and the small section on built-in types.

### CODE REVIEW

File: manual/manual/refman/const.etex [hunk 1]
```diff
@@ -9,6 +9,9 @@
 \begin{syntax}
 constant:
     integer-literal
+  | int32-literal
+  | int64-literal
+  | native-literal
```
Comment (gasche) [inline manual/manual/refman/const.etex]: This should be `nativeint-literal`

File: manual/manual/library/builtin.etex [hunk 2]
```diff
@@ -103,6 +103,7 @@ type int32
 \index{int32@\verb`int32`}
 \begin{ocamldocdescription}
  The type of signed 32-bit integers.
+ Literals for 32-bit integers are suffixed by l.
```
Comment (gasche) [inline manual/manual/library/builtin.etex:106]: This needs not be part of this PR, but it would be user-friendly to give examples of literal values at each of those described types: `()`, `Exit`, `[1; 2; 3]`, `Some true`, `1l, -1l`, etc.

**Review by gasche — APPROVED** [review]
Approved, but see minor typo in comments.

File: manual/manual/refman/const.etex (hunk 1 above)
Comment (Octachron) [reply manual/manual/refman/const.etex]: Fixed, thanks!

File: manual/manual/library/builtin.etex (hunk 2 above)
Comment (Octachron) [reply manual/manual/library/builtin.etex:106]: It sounds like a good idea, but for a separate PR.
